pkg/catalogv2/system/system.go

```diff
@@ -268,12 +289,10 @@ func (m *Manager) install(namespace, name, minVersion, exactVersion string, valu
 		return nil
 	}
 
-	if ok, err := m.hasStatus(namespace, name, action.ListPendingInstall); err != nil {
+	if ok, err := m.hasStatus(namespace, name, action.ListPendingInstall|action.ListPendingUpgrade); err != nil {
```

Comment:
I think we should bring back the purpose of uninstalling a helm chart when it is stuck in `pending` like this commit was supposed to do https://github.com/rancher/rancher/commit/f08915f3c706f565238bb1f921b45b243175258f.

Can we copy the uninstall part of the codebase add it to here inside `else if ok`  ?

Then, I also think we need to increase the time from 60 sec to maybe 2 3 4 5 min in the waitPodDone func ? 

Thinking about the commit https://github.com/rancher/rancher/commit/f08915f3c706f565238bb1f921b45b243175258f, the esclation we had where the new eks-operator upgrade failed wouldn't have happened if this was there. 

But the logic in that commit was removed by a refactoring commit - https://github.com/rancher/rancher/pull/32996 which might have been done by mistake ? Which gives us the more reason to add tests so that things like these won't happen in future.

Reply:
This has been solved. The commit https://github.com/rancher/rancher/commit/f08915f3c706f565238bb1f921b45b243175258f is no longer needed now since there is helm timeout.